fat/fat2/transaction_test.go

```diff
@@ -0,0 +1,317 @@
+		"outputs": [{"address": "FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC","type": "PEG"}]
+	}`,
+	ExpectedDeltas: `{
+		"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q": {"PEG": -50},
+		"FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC": {"PEG": 50}
+	}`,
+}, {
+	Name: "valid transfer (implicit output type & amount)",
+	TxJSON: `{
+		"input": {"address": "FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q", "type": "PEG", "amount": 50},
+		"outputs": [{"address": "FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC"}]
+	}`,
+	ExpectedDeltas: `{
+		"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q": {"PEG": -50},
+		"FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC": {"PEG": 50}
+	}`,
+}, {
+	Name: "valid transfer (overflow input stays in input address)",
+	TxJSON: `{
+		"input": {"address": "FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q", "type": "PEG", "amount": 50},
+		"outputs": [{"address": "FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC", "type": "PEG", "amount": 25}]
+	}`,
+	ExpectedDeltas: `{
+		"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q": {"PEG": -25},
+		"FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC": {"PEG": 25}
+	}`,
+}, {
+	Name: "valid transfer (send to self)",
+	TxJSON: `{
+		"input": {"address": "FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q", "type": "PEG", "amount": 502},
+		"outputs": [{"address": "FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q", "type": "PEG", "amount": 50}]
+	}`,
+	ExpectedDeltas: `{
+		"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q": {"PEG": 0}
+	}`,
+}, {
+	Name:  "not enough input to cover outputs",
+	Error: "insufficient input to cover outputs",
+	TxJSON: `{
+		"input": {"address": "FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q", "type": "PEG", "amount": 50},
+		"outputs": [
+			{"address": "FA1zT4aFpEvcnPqPCigB3fvGu4Q4mTXY22iiuV69DqE1pNhdF2MC", "type": "PEG", "amount": 50},
+			{"address": "FA3tmyjSCo5hPHZn12GGBEMjrvK4ZpVv5XV1nuL2Rd1jgTjVTuJX", "type": "PEG", "amount": 50}
+		]
+	}`,
+}}
+
+// TestTransaction_GetBalanceDeltas essentially tests that the execution of a transaction behaves as expected. That
+// is to say that the transaction will subtract the expected amount of funds from the input address, add the
+// expected amount of funds to the output addresses, and raise an error when it would be impossible to satisfy outputs
+// with explicitly stated inputs.
+//
+// TODO: test conversions behave as expected as well
+func TestTransaction_GetBalanceDeltas(t *testing.T) {
+	conversionRates := make(map[string]float64)
+	for _, test := range transactionGetDeltasTests {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			var tx Transaction
+			err := json.Unmarshal([]byte(test.TxJSON), &tx)
+			assert.Nilf(err, "Transaction JSON Unmarshal raised an unexpected error")
+			err = tx.Valid()
+			assert.Nilf(err, "Transaction.Valid() raised an unexpected error")
```

Comment:
use require.NoError and reduce the message to "Transaction.Valid()". that better leverages the testify packages as it will then tell you that it wants no error.

Reply:
Done